content/source/docs/extend/testing/acceptance-tests/teststep.html.md

```diff
@@ -0,0 +1,291 @@
+configuration, based on a randomized name provided, however an in-line string or
+constant string would work as well, so long as they contain valid Terraform
+configuration for the plugin or resource under test. This pattern of first
+applying and checking a basic configuration, followed by applying a modified
+configuration with updated or additional checks is a common pattern used to test
+update functionality.
+
+
+## Check Functions
+
+After the configuration for a `TestStep` is applied, Terraform’s testing
+framework provides developers an opportunity to check the results by providing a
+“Check” function. While possible to only supply a single function, it is
+recommended you use multiple functions to validate specific information about
+the results of the `terraform apply` ran in each `TestStep`. The `Check`
+attribute is of `TestStep` is singular, so in order to include multiple checks
+developers should use either `ComposeTestCheckFunc` or
+`ComposeAggregateTestCheckFunc` (defined below) to group multiple check
+functions, defined below:
+
+### ComposeTestCheckFunc 
+
+ComposeTestCheckFunc lets you compose multiple TestCheckFunc functions into a
+single check. As a user testing their provider, this lets you decompose your
+checks into smaller pieces more easily, with individual methods for checking
+specific attributes. Each check is ran in the order provided, and on failure the
+entire `TestCase` is stopped, and Terraform attempts to destroy any resources
+created.
+
+Example:
+
+```go 
+Steps: []resource.TestStep{
+  {
+    Config: testAccExampleResource(rName),
+    Check: resource.ComposeTestCheckFunc(
+		// if testAccCheckExampleResourceExists fails to find the resource, 
+		// the parent TestStep and TestCase fail
+      testAccCheckExampleResourceExists("example_widget.foo", &widgetBefore), 
+      resource.TestCheckResourceAttr("example_widget.foo", "size", "expected size"),
+    ),
+  },
+},
+```
+
+### ComposeAggregateTestCheckFunc 
+
+ComposeAggregateTestCheckFunc lets you compose multiple TestCheckFunc functions
+into a single check. It’s purpose and usage is identical to
+ComposeTestCheckFunc, however each check is ran in order even if a previous
+check failed, collecting the errors returned from any checks and returning a
+single aggregate error. The entire `TestCase` is still stopped, and Terraform
+attempts to destroy any resources created. 
+
+Example:
+
+```
+Steps: []resource.TestStep{
+  {
+    Config: testAccExampleResource(rName),
+    Check: resource.ComposeAggregateTestCheckFunc(
+      testAccCheckExampleResourceExists("example_widget.foo", &widgetBefore), // if testAccCheckExampleResourceExists fails to find the resource, the following TestCheckResourceAttr is still ran, with any errors aggregated
+      resource.TestCheckResourceAttr("example_widget.foo", "active", "true"),
+    ),
+  },
+},
+```
+
+## Builtin check functions
+
+Terraform has several TestCheckFunc functions builtin for developers to use for
```

Comment:
Instinctively I think "builtin" should be "built in" in this sentence but I may be wrong.